mifosng-provider/src/main/java/org/mifosplatform/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -194,6 +204,23 @@ private CommandProcessingResult createGroupingType(final JsonCommand command, fi
              * groups if they exist
              */
             newGroup.generateHierarchy();
+            
+            if (newGroup.isAccountNumberRequiresAutoGeneration()) {
+            	EntityAccountType entityAccountType = null;
```

Comment:
Move this block of code into a new method say:

generateAccountNumberIfRequired(Group group)
Ideally, we would want to keep the size of a method less than 50 lines of code (I know it is not done currently, but we would be doing a round of refactoring soon)